packages/postgres-database/tests/test_groups.py

```diff
@@ -21,233 +24,234 @@
-        assert beatles_group.modified > quarrymen_group.modified
-
-        # delete 1 user
-        await conn.execute(users.delete().where(users.c.id == pete.id))
-
-        # check DB contents
-        users_count = await conn.scalar(select([func.count()]).select_from(users))
-        assert users_count == 4
-        groups_count = await conn.scalar(select([func.count()]).select_from(groups))
-        assert groups_count == (users_count + 2 + 1)
-        relations_count = await conn.scalar(
-            select([func.count()]).select_from(user_to_groups)
-        )
-        assert relations_count == (users_count + users_count + users_count)
-
-        # add one user to another group
-        await conn.execute(
-            user_to_groups.insert().values(uid=ringo.id, gid=beatles_group.gid)
-        )
-
-        # check DB contents
-        users_count = await conn.scalar(select([func.count()]).select_from(users))
-        assert users_count == 4
-        groups_count = await conn.scalar(select([func.count()]).select_from(groups))
-        assert groups_count == (users_count + 2 + 1)
-        relations_count = await conn.scalar(
-            select([func.count()]).select_from(user_to_groups)
-        )
-        assert relations_count == (users_count + users_count + 1 + users_count)
-
-        # delete 1 group
-        await conn.execute(groups.delete().where(groups.c.gid == rory_group.gid))
-
-        # check DB contents
-        users_count = await conn.scalar(select([func.count()]).select_from(users))
-        assert users_count == 4
-        groups_count = await conn.scalar(select([func.count()]).select_from(groups))
-        assert groups_count == (users_count + 1 + 1)
-        relations_count = await conn.scalar(
-            select([func.count()]).select_from(user_to_groups)
-        )
-        assert relations_count == (users_count + users_count + users_count)
-
-        # delete the other group
-        await conn.execute(groups.delete().where(groups.c.gid == beatles_group.gid))
-
-        # check DB contents
-        users_count = await conn.scalar(select([func.count()]).select_from(users))
-        assert users_count == 4
-        groups_count = await conn.scalar(select([func.count()]).select_from(groups))
-        assert groups_count == (users_count + 0 + 1)
-        relations_count = await conn.scalar(
-            select([func.count()]).select_from(user_to_groups)
-        )
-        assert relations_count == (users_count + users_count)
+    rory_group = await create_fake_group(
+        connection, name="Rory Storm and the Hurricanes"
+    )
+    quarrymen_group = await create_fake_group(connection, name="The Quarrymen")
+    await create_fake_user(connection, name="John", group=quarrymen_group)
+    await create_fake_user(connection, name="Paul", group=quarrymen_group)
+    await create_fake_user(connection, name="Georges", group=quarrymen_group)
+    pete = await create_fake_user(connection, name="Pete", group=quarrymen_group)
```

Comment:
hope you learned something about the beatles here...

Reply:
:-D your tests were very educational :-)